tests/components/dyson/test_air_quality.py

```diff
@@ -0,0 +1,130 @@
+            {
+                'device_id': 'XX-XXXXX-XX',
+                'device_ip': '192.168.0.1'
+            }
+        ]
+    }}
+
+
+@asynctest.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+@asynctest.patch('libpurecool.dyson.DysonAccount.devices',
+                 return_value=[_get_dyson_purecool_device()])
+async def test_purecool_aiq_attributes(devices, login, hass):
+    """Test state attributes."""
+    await async_setup_component(hass, dyson_parent.DOMAIN, _get_config())
+    await hass.async_block_till_done()
+    fan_state = hass.states.get("air_quality.living_room")
+    attributes = fan_state.attributes
+
+    assert fan_state.state == '42'
+    assert attributes[dyson.ATTR_PM_2_5] == 14
+    assert attributes[dyson.ATTR_PM_10] == 25
+    assert attributes[dyson.ATTR_NO2] == 42
+    assert attributes[dyson.ATTR_VOC] == 35
+
+
+@asynctest.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+@asynctest.patch('libpurecool.dyson.DysonAccount.devices',
+                 return_value=[_get_dyson_purecool_device()])
+async def test_purecool_aiq_update_state(devices, login, hass):
+    """Test state update."""
+    device = devices.return_value[0]
+    await async_setup_component(hass, dyson_parent.DOMAIN, _get_config())
+    await hass.async_block_till_done()
+    event = {
+        "msg": "ENVIRONMENTAL-CURRENT-SENSOR-DATA",
+        "time": "2019-03-29T10:00:01.000Z",
+        "data": {
+            "pm10": "0080",
+            "p10r": "0151",
+            "hact": "0040",
+            "va10": "0055",
+            "p25r": "0161",
+            "noxl": "0069",
+            "pm25": "0035",
+            "sltm": "OFF",
+            "tact": "2960"
+        }
+    }
+    device.environmental_state = \
+        DysonEnvironmentalSensorV2State(json.dumps(event))
+
+    callback = device.add_message_listener.call_args_list[2][0][0]
+    callback(device.state)
+    await hass.async_block_till_done()
+    fan_state = hass.states.get("air_quality.living_room")
+    attributes = fan_state.attributes
+
+    assert fan_state.state == '80'
+    assert attributes[dyson.ATTR_PM_2_5] == 35
+    assert attributes[dyson.ATTR_PM_10] == 80
+    assert attributes[dyson.ATTR_NO2] == 69
+    assert attributes[dyson.ATTR_VOC] == 55
+
+
+@asynctest.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+@asynctest.patch('libpurecool.dyson.DysonAccount.devices',
+                 return_value=[_get_dyson_purecool_device()])
+async def test_purecool_component_setup_only_once(devices, login, hass):
+    """Test if entities are created only once."""
+    config = _get_config()
+    await async_setup_component(hass, dyson_parent.DOMAIN, config)
+    await hass.async_block_till_done()
+    discovery.load_platform(hass, "air_quality",
+                            dyson_parent.DOMAIN, {}, config)
+    await hass.async_block_till_done()
+
+    aiq = [aiq for aiq in hass.data[AIQ_DOMAIN].entities
```

Comment:
Don't check the component entities. If needed check platform saved entities in hass.data.

Reply:
Done.